pkg/deploy/registry/deployconfig/etcd/etcd.go

```diff
@@ -46,3 +75,150 @@ func NewStorage(s storage.Interface) *REST {
 
 	return &REST{store}
 }
+
+// ScaleREST contains the REST storage for the Scale subresource of DeploymentConfigs.
+type ScaleREST struct {
+	registry     *deployconfig.Registry
```

Comment:
a pointer to an interface is usually wrong.  Do you really need or mean it here?  It seems to have led to unexpected dereferences.